### PR TITLE
Use managed identity for sql ad auth. Removes need for credentials

### PR DESCRIPTION
**Pull request by bbarber**
I added a couple links in the main `README.md`
image

Here are the links to those files. Viewing them in "Preview" mode is way easier to review imo

https://github.com/dontpaniclabs/SolutionTemplate/blob/bbarber/managed-identity-remove-sql-credentials/templates/dplsln/Infrastructure/README.md

https://github.com/dontpaniclabs/SolutionTemplate/blob/bbarber/managed-identity-remove-sql-credentials/templates/dplsln/Pipelines/README.md

### CODE REVIEW

File: templates/dplsln/BackendServices/DPL.Template.Accessors/EntityFramework/partials/DatabaseContext.cs [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
+using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 
 namespace DPL.Template.Accessors.EntityFramework
 {
     public partial class DatabaseContext : DbContext
     {
-        protected IConfigurationRoot Configuration { get; set; }
-
+        [SuppressMessage("Microsoft.Design", "CA2000", Justification = "DbContext has its own using statement")]
```
Comment (bbarber) [inline templates/dplsln/BackendServices/DPL.Template.Accessors/EntityFramework/partials/DatabaseContext.cs:12]: Can you fact check me on this? we're `using var db new DatabaseContext()` in our base accessor. Is this memory tight?

File: templates/dplsln/Infrastructure/azuredeploy.json [hunk 2]
```diff
@@ -26,17 +26,17 @@
...
       "type": "String",
       "metadata": {
         "description": "Connection string name as found in appsettings.json (Example: 'DPLDatabase')"
       }
+    },
+    "sql_server_sa_password": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Initial database password. User defaults to server name. This password is required for server creation, but is not used otherwise"
+      }
```
Comment (bbarber) [inline templates/dplsln/Infrastructure/azuredeploy.json:39]: Password is required for sql server creation, but is not used anywhere else. I wish we could completely remove this admin account, but I don't think it's possible

**Comment by bbarber** [comment]
I need to fix the preprocessor directives, as these will get processed and removed using Release (essentially removing the Debug code). Marking as draft until this is resolved

File: templates/dplsln/BackendServices/DPL.Template.Accessors/EntityFramework/partials/DatabaseContext.cs (hunk 1 above)
Comment (bbarber) [reply templates/dplsln/BackendServices/DPL.Template.Accessors/EntityFramework/partials/DatabaseContext.cs]: Resolved

**Review by twilsman — APPROVED** [review]

File: DPLTemplates.csproj [hunk 3]
```diff
@@ -12,11 +12,12 @@
     <NoWarn>$(NoWarn);NU5128</NoWarn>
     <IncludeContentInPack>true</IncludeContentInPack>
     <IncludeBuildOutput>false</IncludeBuildOutput>
+    <NoDefaultExcludes>true</NoDefaultExcludes>
     <ContentTargetFolders>content</ContentTargetFolders>
   </PropertyGroup>
 
   <ItemGroup>
-    <Content Include="templates\**\*" Exclude="templates\**\bin\**;templates\**\obj\**" />
+    <Content Include="templates\**\*" Exclude="templates\**\bin\**;templates\**\obj\**;templates\**\node_modeules\**" />
```
Comment (twilsman) [inline DPLTemplates.csproj:20]: `modeules` is misspelled.

Comment (bbarber) [reply DPLTemplates.csproj]: Fixed!

**Pull request merged**